featurebyte/query_graph/pruning_util.py

```diff
@@ -27,7 +29,6 @@ def get_prune_graph_and_nodes(feature_objects: list[FeatureModel]) -> tuple[Quer
     local_query_graph = QueryGraph()
     feature_nodes = []
     for feature in feature_objects:
-        pruned_graph, mapped_node = feature.extract_pruned_graph_and_node()
-        local_query_graph, local_name_map = local_query_graph.load(pruned_graph)
-        feature_nodes.append(local_query_graph.get_node_by_name(local_name_map[mapped_node.name]))
+        local_query_graph, local_name_map = local_query_graph.load(feature.graph)
+        feature_nodes.append(local_query_graph.get_node_by_name(local_name_map[feature.node_name]))
     return local_query_graph, feature_nodes
```

Comment:
Use the following function to benchmark performance:

```
def feature_list_creation(feature_docs):
    start = time.time()
    features = []
    feature_ids = []
    for feature_doc in feature_docs:
        feature = FeatureModel(**feature_doc)
        features.append(feature)
        feature_ids.append(feature.id)

    FeatureListModel(features=features, feature_ids=feature_ids, version="V231110")
    elapsed = time.time() - start
    return elapsed
```

When convert 721 feature documents to feature models & create feature list model,
current method takes ~5.54s, revised method takes 1.76s.

Reply:
Since the stored feature model has already stored pruned graph, we could skip graph pruning step when constructing feature clusters.